### PR TITLE
Client and Storage rework: Recognize multiple API groups

### PR DESCRIPTION
**Pull request by luxas**
Before this, the client and storage framework was locked in to only one particular combination of `apiVersion`, or in other words, `GroupVersion`. We only checked `meta.Kind` everywhere; but that always implicitly assumed that the GroupVersion was ignite/v1alpha1

Fixes: #199 

This PR:
 - Makes the `Storage` (`Cache`), and `Filterer` interfaces operate on `schema.GroupVersionKind`, not `meta.Kind`
 - Automatically sets the desired GVK when decoding or creating an object, so the hacky defaulting functions we had are now obsolete
 - Extracts ignite-related stuff from the `Client` struct to a gv-specific `IgniteInternalClient`, and passes through the GVK automatically to the storage as a consequence.
 - Adds a `Client.New` -> `Storage.New` method being the canonical way to create a new API object. In other words, do not do `&api.VM{}` or similar anymore.
 - Adapts the rest of the application to this new code flow

### CODE REVIEW

File: pkg/client/client_resource_template.go [hunk 1]
```diff
@@ -9,15 +9,21 @@
...
 
 import (
+	"fmt"
+
 	log "github.com/sirupsen/logrus"
 	api "github.com/weaveworks/ignite/pkg/apis/ignite"
 	meta "github.com/weaveworks/ignite/pkg/apis/meta/v1alpha1"
 	"github.com/weaveworks/ignite/pkg/storage"
 	"github.com/weaveworks/ignite/pkg/storage/filterer"
+
```
Comment (twelho) [inline pkg/client/client_resource_template.go]: Remove blank line?

File: pkg/client/zz_generated.client_image.go [hunk 2]
```diff
@@ -8,15 +8,21 @@
...
 
 import (
+	"fmt"
+
 	log "github.com/sirupsen/logrus"
 	api "github.com/weaveworks/ignite/pkg/apis/ignite"
 	meta "github.com/weaveworks/ignite/pkg/apis/meta/v1alpha1"
 	"github.com/weaveworks/ignite/pkg/storage"
 	"github.com/weaveworks/ignite/pkg/storage/filterer"
+
```
Comment (twelho) [inline pkg/client/zz_generated.client_image.go]: Rerun `make tidy` after removing the blank line from template

File: pkg/storage/cache.go [hunk 3]
```diff
@@ -2,8 +2,9 @@ package storage
 
 import (
 	log "github.com/sirupsen/logrus"
-
 	meta "github.com/weaveworks/ignite/pkg/apis/meta/v1alpha1"
+
```
Comment (twelho) [inline pkg/storage/cache.go]: Remove two blank lines

File: pkg/storage/filterer/filterer.go [hunk 4]
```diff
@@ -5,6 +5,8 @@ import (
 
 	meta "github.com/weaveworks/ignite/pkg/apis/meta/v1alpha1"
 	"github.com/weaveworks/ignite/pkg/storage"
+
```
Comment (twelho) [inline pkg/storage/filterer/filterer.go]: Remove blank line

**Comment by luxas** [comment]
The blank lines are rebase conflict. I'll fixup those in this follow up: #224

**Pull request merged**